sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,593 @@
+   * @param rule
+   * @param ruleOptions
+   */
+  async updateRule(
+    topicPath: string,
+    subscriptionPath: string,
+    rule: string,
+    ruleOptions: RuleOptions
+  ): Promise<RuleResult> {
+    const fullPath = this.getRulePath(topicPath, subscriptionPath, rule);
+
+    const response: HttpOperationResponse = await this._putResource(
+      fullPath,
+      ruleOptions,
+      true,
+      this.ruleResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Deletes a rule.
+   * @param topicPath
+   * @param subscriptionPath
+   * @param rule
+   */
+  async deleteRule(topicPath: string, subscriptionPath: string, rule: string): Promise<RuleResult> {
+    const fullPath = this.getRulePath(topicPath, subscriptionPath, rule);
+    const response: HttpOperationResponse = await this._deleteResource(
+      fullPath,
+      this.ruleResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Gets a rule.
+   * @param topicPath
+   * @param subscriptionPath
+   * @param rule
+   */
+  async getRule(topicPath: string, subscriptionPath: string, rule: string): Promise<RuleResult> {
+    const fullPath = this.getRulePath(topicPath, subscriptionPath, rule);
+    const response: HttpOperationResponse = await this._getResource(
+      fullPath,
+      this.ruleResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Lists existing rules.
+   * @param topicPath
+   * @param subscriptionPath
+   * @param listRequestOptions
+   */
+  async listRules(
+    topicPath: string,
+    subscriptionPath: string,
+    listRequestOptions?: ListRequestOptions
+  ): Promise<ListRulesResult> {
+    const fullPath = this.getSubscriptionPath(topicPath, subscriptionPath) + "/Rules/";
+    const response: HttpOperationResponse = await this._listResources(
+      fullPath,
+      listRequestOptions,
+      this.ruleResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * @param queuePath
+   */
+  formatDeadLetterPath(queuePath: string): string {
+    return `${queuePath}/$DeadLetterQueue`;
+  }
+
+  /**
+   * Creates or updates a resource based on `isUpdate` parameter.
+   * @param path
+   * @param entityFields
+   * @param isUpdate
+   * @param serializer
+   */
+  private async _putResource(
+    path: string,
+    entityFields: QueueOptions | TopicOptions | SubscriptionOptions | RuleOptions,
+    isUpdate: boolean = false,
+    serializer: AtomXmlSerializer
+  ): Promise<HttpOperationResponse> {
+    const webResource: WebResource = new WebResource(this.getUrl(path), "PUT");
+    webResource.body = JSON.stringify(entityFields);
+    if (isUpdate) {
+      webResource.headers.set("If-Match", "*");
+    }
+    webResource.headers.set("content-type", "application/atom+xml;type=entry;charset=utf-8");
+    webResource.headers.set("content-length", Buffer.byteLength(webResource.body));
+
+    const atomXmlOperationSpec: AtomXmlOperationSpec = {
```

Comment:
The constant not required as it gets used only once. Please consider inlining the setting of the operation spec. 

Same for get, list and delete resource methods